python/ray/tune/tests/test_tuner_restore.py

```diff
@@ -713,7 +713,7 @@ def create_trainable_with_params():
         )
         return trainable_with_params
 
-    exp_name = "restore_with_params"
+    exp_name = f"restore_with_params-{use_function_trainable=}"
```

Comment:
This change was required to avoid a bug where:
* Both tests would use the same exp name and save things to the same driver staging dir
* `tuner.pkl` gets downloaded to the staging dir when the test runs the first time with the `use_function_trainable=True`. 
* Then, when the second run happens, it writes the correct `tuner.pkl` to storage, but then does another driver sync where the old `tuner.pkl` that was downloaded gets uploaded again.
* This caused an error when restoring the 2nd run.

This problem gets fixed in the follow-up PR by using unique staging directories for each new ray train experiment.

Reply:
Users should never run Tune twice in the same job? 
I think this will no longer be a problem after your second PR, which separates driver staging directories with timestamp.